frontend/src/components/Widgets/CustomEntitiesDropdown.tsx

```diff
@@ -721,7 +744,9 @@ export default function CustomEntitiesDropdown({
                                   disableRipple
                                   icon={<UncheckedItemIcon fontSize="small" />}
                                   checkedIcon={
-                                    <CheckedItemIcon fontSize="small" />
+                                    (areArraysOfObjectsEqual(selectedOptions, preLevelItems))
```

Comment:
Might be better to check if the arrays are equal just once and store that value in a variable, instead of doing that comparation every time you need